storage/google/cloud/storage/_helpers.py

```diff
@@ -71,6 +72,14 @@ def user_project(self):
         """Abstract getter for the object user_project."""
         raise NotImplementedError
 
+    @property
+    def _query_params(self):
+        """Default query parameters."""
+        params = {}
+        if self.user_project is not None:
+            params["userProject"] = self.user_project
+        return params
```

Comment:
This method is shadowed by both `Blob` and `Bucket`:  let's make it virtual, e.g.:

```python
    @property
    def _query_params(self):
        """Default query parameters."""
        raise NotImplementedError
```

That way we don't have to add test coverage for it, either.

Reply:
Looking at how it is used in `reload`, I've changed my mind:  lets just get rid of the duplicate version in `Bucket`.